staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -107,15 +110,25 @@ type DiffOptions struct {
 	FieldManager    string
 	ForceConflicts  bool
 
-	Selector         string
-	OpenAPISchema    openapi.Resources
-	DiscoveryClient  discovery.DiscoveryInterface
-	DynamicClient    dynamic.Interface
-	DryRunVerifier   *resource.DryRunVerifier
-	CmdNamespace     string
-	EnforceNamespace bool
-	Builder          *resource.Builder
-	Diff             *DiffProgram
+	Selector          string
+	OpenAPISchema     openapi.Resources
+	DiscoveryClient   discovery.DiscoveryInterface
+	DynamicClient     dynamic.Interface
+	DryRunVerifier    *resource.DryRunVerifier
+	CmdNamespace      string
+	EnforceNamespace  bool
+	Builder           *resource.Builder
+	Diff              *DiffProgram
+	Mapper            meta.RESTMapper
+	Prune             bool
+	PruneResources    []pruneResource
+	VisitedUids       sets.String
+	VisitedNamespaces sets.String
+	ToPrinter         func(string) (printers.ResourcePrinter, error)
+	PrintFlags        *genericclioptions.PrintFlags
```

Comment:
Why do we need these? How are you planning on showing what was deleted? By showing "deleted" in the output?

Reply:
Before showing the differences in file, if `--prune` flag is passed, it'll show;
```
namespace/test pruned
deployment/test pruned
...
```

Reply:
Yeah, that's highly inconsistent with how diff works. If you want to see these, I would suggest you run `kubectl apply --dry-run=client`. The output of kubectl diff should be something that can be diffed. In that specific instance, I would expect the current object to be put in the "old" folder, and no object to be put in the "new" folder, which will force `diff -RN` to show the object as removed.